src/cmds/mod.rs

```diff
@@ -0,0 +1,525 @@
+    AddGwmpRegion(AddGwmpRegion),
+    /// Remove a region mapping from the Gwmp Protocol.
+    /// This only works if the protocol is already gwmp.
+    RemoveGwmpRegion(RemoveGwmpRegion),
+    /// Set the Route Protocol to PacketRouter (GRPC)
+    PacketRouter(UpdatePacketRouter),
+}
+
+#[derive(Debug, Args)]
+pub struct UpdateMaxCopies {
+    #[arg(short, long)]
+    pub route_id: String,
+    #[arg(short, long)]
+    pub max_copies: u32,
+    #[arg(from_global)]
+    pub keypair: PathBuf,
+    #[arg(from_global)]
+    pub config_host: String,
+    #[arg(long)]
+    pub commit: bool,
+}
+
+#[derive(Debug, Args)]
+pub struct UpdateServer {
+    #[arg(short, long)]
+    pub route_id: String,
+    #[arg(long)]
+    pub host: String,
+    #[arg(long)]
+    pub port: u32,
+    #[arg(from_global)]
+    pub keypair: PathBuf,
+    #[arg(from_global)]
+    pub config_host: String,
+    #[arg(long)]
+    pub commit: bool,
+}
+
+#[derive(Debug, Args)]
+pub struct UpdateHttp {
+    #[arg(short, long)]
+    pub route_id: String,
+    #[arg(short, long, default_value = "250")]
+    pub dedupe_timeout: u32,
+    /// Just the path part of the Server URL
+    ///
+    /// The rest will be taken from the Server {host}:{port}
+    #[arg(short, long)]
+    pub path: String,
+    /// Authorization Header
+    #[arg(short, long)]
+    pub auth_header: Option<String>,
+
+    #[arg(from_global)]
+    pub keypair: PathBuf,
+    #[arg(from_global)]
+    pub config_host: String,
+    #[arg(long)]
+    pub commit: bool,
+}
+
+#[derive(Debug, Args)]
+pub struct UpdatePacketRouter {
+    #[arg(short, long)]
+    pub route_id: String,
+    #[arg(from_global)]
+    pub keypair: PathBuf,
+    #[arg(from_global)]
+    pub config_host: String,
+    #[arg(long)]
+    pub commit: bool,
+}
+
+#[derive(Debug, Args)]
+pub struct AddGwmpRegion {
+    #[arg(short, long)]
+    pub route_id: String,
+    #[arg(value_enum)]
+    pub region: Region,
+    pub region_port: u32,
+
+    #[arg(from_global)]
+    pub keypair: PathBuf,
+    #[arg(from_global)]
+    pub config_host: String,
+    #[arg(long)]
+    pub commit: bool,
+}
+
+#[derive(Debug, Args)]
+pub struct RemoveGwmpRegion {
+    #[arg(short, long)]
+    pub route_id: String,
+    #[arg(value_enum)]
+    pub region: Region,
+
+    #[arg(from_global)]
+    pub keypair: PathBuf,
+    #[arg(from_global)]
+    pub config_host: String,
+    #[arg(long)]
+    pub commit: bool,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum EuiCommands {
+    /// Get all EUI pairs for a Route
+    List(ListEuis),
+    /// Add EUI pair to Route
+    Add(AddEui),
+    /// Remove EUI pair from Route
+    Remove(RemoveEui),
+    /// Remove ALL EUI Pairs from Route
+    Clear(ClearEuis),
+}
+
+#[derive(Debug, Subcommand)]
+pub enum DevaddrCommands {
+    /// Get all Devaddr Ranges for a Route
+    List(ListDevaddrs),
+    /// Add Devaddr Range to Route
+    Add(AddDevaddr),
+    /// Remove Devaddr Range from Route
+    Delete(DeleteDevaddr),
```

Comment:
should this be `Remove` as well to mirror the eui pair subcommand?

Reply:
Yes it should. Thanks for that catch.
This was leftover from the previous command structure.